### ISSUE TITLE
restore state appears to attempt to Kill twice

### ISSUE THREAD

**Issue by macb** [hashicorp/nomad#427]
Figured I'd give the new state reset merges a try and everything worked great. However it looks like the cleanup might run twice though, leading to slightly confusing log lines.

```
2015/11/17 01:46:27 [INFO] driver.docker: re-attaching to docker process: DOCKER:{"ImageID":"2899e038764feba44a8c0d2a99c1793e6a67b7a1d800138c9b240e93e5cb9d8a","ContainerID":"96e8c3128199ea076416e50de7880a38461a11ec350f433925caed1e7bb65317"}
2015/11/17 01:46:33 [INFO] driver.docker: stopped container 96e8c3128199ea076416e50de7880a38461a11ec350f433925caed1e7bb65317
2015/11/17 01:46:33 [INFO] driver.docker: removed container 96e8c3128199ea076416e50de7880a38461a11ec350f433925caed1e7bb65317
2015/11/17 01:46:33 [INFO] driver.docker: removed image 2899e038764feba44a8c0d2a99c1793e6a67b7a1d800138c9b240e93e5cb9d8a
2015/11/17 01:46:33 [ERR] driver.docker: failed stopping container 96e8c3128199ea076416e50de7880a38461a11ec350f433925caed1e7bb65317
2015/11/17 01:46:33 [ERR] client: failed to kill task 'repeater' for alloc 'be907446-ec27-5b95-857f-ef6c5a3a0c0d': Failed to stop container 96e8c3128199ea076416e50de7880a38461a11ec350f433925caed1e7bb65317: No such container: 96e8c3128199ea076416e50de7880a38461a11ec350f433925caed1e7bb65317
```

**Comment by dadgar** [comment]
Just verified the bug. Thanks for reporting!

**Comment by dadgar** [comment]
Should be fixed by baeea15aa2b8f4fc498f840d85159b7af8b10d4f
